src/Extensions/Hosting.Services.Web/Middlewares/UserIdentity/UserIdentityMiddleware.cs

```diff
@@ -57,7 +58,7 @@ internal async Task<string> CreateUserHashAsync(HttpContext context)
 			foreach (IUserIdentityProvider provider in m_userIdentityProviders)
 			{
 				bool success;
-				(success, identityBytesWritten) = await provider.TryWriteBytesAsync(context, uidMemoryOwner.Memory.Span.Slice(0, provider.MaxBytesInIdentity))
+				(success, identityBytesWritten) = await provider.TryWriteBytesAsync(context, uidMemoryOwner.Memory)
```

Comment:
Since we how giving identity providers access to the whole memory we probably should call `.Fill(0)` (similar to how it's done above) when `success` is `false`. To avoid the case when identity provider filled memory with something and returned `false`.